standards/core/sync.md

```diff
@@ -7,126 +7,204 @@ contributors:
-for the range of time
-during which synchronization is required.
-Nodes SHOULD use the same time range,
-for Waku we chose one hour as the global default.
-Waku Relay or Light Push protocol MAY be enabled
-and used in conjunction with Sync
-as a source of new message hashes
-for the cache.
-
-Nodes MAY use the Store protocol
-to request missing messages once reconciliation is complete
-or to provide messages to requesting clients.
-
-### Payload
+## Reconciliation
 
+**Libp2p Protocol identifier**: `/vac/waku/reconciliation/1.0.0`
+
+The protocol finds differences between 2 peers by
+comparing _fingerprints_ of _ranges_ of message _IDs_.
+_Ranges_ are encoded into payloads, exchanged between the peers and when the range _fingerprints_ are different, splitted into smaller ones.
+This process repeats until _ranges_ include a small number of messages
+at this point _IDs_ are sent for comparison instead of _fingerprints_.
+
+#### Overview
+
+1. The requestor choose a time range to sync.
+2. The range is encoded into a payload and sent.
+3. The requestee receive the payload and decode it.
+4. The range is processed and more ranges produced.
+5. The new ranges are encoded and sent.
+6. Payloads are repeatedly exchanged and differences between the peers are discovered.
+7. The synchronization is done when no ranges are left to process.
+
+#### Message Ids
+Message _IDs_ MUST be composed of the timestamp and the hash of the [`14/WAKU2-MESSAGE`](https://rfc.vac.dev/waku/standards/core/14/message).
+
+The timestamp MUST be the time of creation and
+the hash MUST follow the
+[deterministic message hashing specification](https://rfc.vac.dev/waku/standards/core/14/message#deterministic-message-hashing)
+
+> This way the message Ids can always be totally ordered,
+first chronologically according to the timestamp and then
+disambiguate based on the hash lexical order
+in cases where the timestamp is the same.
+
+#### Range Bounds
+A _range_ MUST consists of 2 _IDs_, the first bound is
+inclusive the second bound exclusive.
+The first bound MUST be strictly smaller than the second one.
+
+#### Range Fingerprinting
+The _fingerprint_ of a range MUST be the XOR operation applied to
+the hash of all message _IDs_ included in that _range_.
+
+#### Range Type
+Every _range_ MUST have one of the following types; _fingerprint_, _skip_ or _item set_.
+
+- _Fingerprint_ type contains a _fingerprint_.
+- _Skip_ type contains nothing and is used to signal already processed _ranges_.
+- _Item set_ type contains message _Ids_ and a _resolved_ boolean.
+> _Item sets_ are an optimization, sending multiple _IDs_ instead of
+recursing further reduce the number of round-trips.
+
+#### Range Processing
+_Ranges_ have to be processed differently according to their types.
+
+- **Equal** _fingerprint ranges_ MUST become _skip ranges_.
+- **Unequal** _fingerprint ranges_ MUST be splitted into smaller ranges. The new type MAY be either _fingerprint_ or _item set_.
```

Comment:
```suggestion
- **Unequal** _fingerprint ranges_ MUST be split into smaller ranges. The new type MAY be either _fingerprint_ or _item set_.
```

Reply:
Shall we add some guidance here as to when the new type is another fingerprint and when it is an item set? We don't need numbers, unless it's hardcoded, but we could just specify that when the new range is "small enough" the individual IDs can be encoded into an item set.

Reply:
Update: you can even point to the implementation section on "item set threshold". Here you can just mention that if number of items in a range fall below _a_ threshold, it becomes an item set rather than just a fingerprint.